### DISCUSSION TITLE
disable interactive PEM pass phrase prompt when importing password protected key?

### DISCUSSION THREAD

**Discussion by wivaku** [spatie/crypto#10] [Q&A]
I have a password protected private key.
When I try to import that key -and I don't provide the password- the import should fail.
Instead: a prompt is shown that asks for the password.

```
[$passwordProtectedPrivateKey, $publicKey] = (new KeyPair())->password($password)->generate();

$test1 = PrivateKey::fromString($passwordProtectedPrivateKey, $password); // success
$test2 = PrivateKey::fromString($passwordProtectedPrivateKey); // without password → should fail
```

For `$test2` a terminal prompt is shown asking for the password:
```
Enter PEM pass phrase:
```

Would expect that an exception is thrown. Anything I can do to make that happen?

**Comment by wivaku** [comment]
BTW. Using phpunit with interaction disabled:
```
phpunit myTest.php --testdox --no-interaction
```

**Comment by ghost** [comment]
I fixed this in this PR https://github.com/spatie/crypto/pull/16 it is in version 2.0.1
